src/mpid/common/self/mpidu_self.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef MPIDU_SELF_H_INCLUDED
+#define MPIDU_SELF_H_INCLUDED
+
+/* message queue within "self"-comms, i.e. MPI_COMM_SELF and all communicators with size of 1. */
+
+struct self_queue_entry {
+    struct MPIR_Request *req;
+    void *buf;
+    MPI_Aint count;
+    MPI_Datatype datatype;
+    int tag;
+    int context_id;
+    struct self_queue_entry *prev, *next;
+};
```

Comment:
This is a lot to add to the request struct. If I remember correctly, this will easily spill use over into another cache line.